sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClient.java

```diff
@@ -54,10 +106,64 @@
  */
 @ServiceClient(builder = TableClientBuilder.class)
 public final class TableClient {
-    final TableAsyncClient client;
 
-    TableClient(TableAsyncClient client) {
-        this.client = client;
+    private static final String DELIMITER_CONTINUATION_TOKEN = ";";
+    private final ClientLogger logger = new ClientLogger(TableClient.class);
+    private final String tableName;
+    private final AzureTableImpl tablesImplementation;
+    private final TransactionalBatchImpl transactionalBatchImplementation;
+    private final String accountName;
+    private final String tableEndpoint;
+    private final HttpPipeline pipeline;
+    private final TableClient transactionalBatchClient;
+
+    private static final int IMMEDIATELY = -1;
+
+    TableClient(String tableName, HttpPipeline pipeline, String serviceUrl, TableServiceVersion serviceVersion,
+                SerializerAdapter tablesSerializer, SerializerAdapter transactionalBatchSerializer) {
+        try {
+            if (tableName == null) {
+                throw new NullPointerException(("'tableName' must not be null to create TableClient."));
+            }
+
+            if (tableName.isEmpty()) {
+                throw new IllegalArgumentException("'tableName' must not be empty to create a TableClient.");
+            }
+
+            final URI uri = URI.create(serviceUrl);
+            this.accountName = uri.getHost().split("\\.", 2)[0];
+            this.tableEndpoint = uri.resolve("/" + tableName).toString();
+            logger.verbose("Table Service URI: {}", uri);
+        } catch (NullPointerException | IllegalArgumentException ex) {
+            throw logger.logExceptionAsError(ex);
+        }
```

Comment:
We could consider moving these validations to a common place that can be shared with the Async client.

Reply:
@samvaity That's certainly possible, but I thought the goal was to change the Public API as little as possible. Extrapolating into a public function within `TableUtils` though under implementation still technically changes the public API if I'm understanding correctly. What's been the cadence for Sync-Stack thus far? How much if at all is it OK to change the Public API?

Reply:
That's correct. We shouldn't be changing the public API at all. 
Moving common code to Utils would still just change the underlying implementation details and not the public API here.

Reply:
@jairmyree Java conventions dictate that anything under an implementation package is not considered part of a library's public API surface, even if the class or method in question uses the `public` keyword.